admin/resources/js/components/H2Components/Button.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+export interface ButtonProps {
+  children: React.ReactNode;
+  borderColor:
+    | "lightpurple"
+    | "lightnavy"
+    | "darkgold"
+    | "white"
+    | "[light]white[0]";
+  backgroundColor:
+    | "lightpurple"
+    | "[light]lightpurple[.1]"
+    | "darkpurple"
+    | "lightnavy"
+    | "[light]lightnavy[.1]"
+    | "darkgold"
+    | "[light]darkgold[.1]"
+    | "white"
+    | "[light]white[0]";
+  fontColor: "black" | "white" | "lightpurple" | "lightnavy" | "darkgold";
```

Comment:
If we're being restrictive on colors, maybe we should just pass in a single color prop, and map that to different combinations.
```
color: "darkpurple" | "lightpurple" | "lightnavy" | "darknavy" | ...;

const colorMap = {
  darkpurple: {
    border: "darkpurple",
    background: "darkpurple",
    font: "white",
  },
  lightpurple: {
     border: "darkpurple",
    background: "lightpurple",
    font: "darkpurple",
  },
  ...
}

Reply:
Or, now that I've looked closer at the stories, use two props:
```
color: "primary" | "secondary" | "cta" | "white;
style: "solid" | "outline" | "inline";

const colorMap = {
  purple_solid: {
    border: "darkpurple",
    background: "darkpurple",
    font: "white",
  },
 ...
}
```
Or, we could even export 12 different Button components, one for each style? 
Either way, if we only want certain combinations, lets set that up in this component.
(BTW, what's CTA?)